### PR TITLE
[14.0] connector_importer: backports from v15 + small improvements

### PR DESCRIPTION
**Pull request by simahawk**
See atomic commits for details.
Commits from [connector_importer: fix source test](https://github.com/OCA/connector-interfaces/commit/c89f42f221babe548a28484130c65cb65a725c14) are new and I'll port compatible ones to v15 when ready.

ref: qoq-1317

### CODE REVIEW

**Review by sebalix — APPROVED** [review]
LG

File: connector_importer/components/importer.py [hunk 1]
```diff
@@ -57,25 +57,22 @@ class RecordImporter(Component):
...
     # log and report errors
     # do not make the whole import fail
     _break_on_error = False
     _record_handler_usage = "odoorecord.handler"
     _tracking_handler_usage = "tracking.handler"
     # a unique key (field name) to retrieve the odoo record
-    # if this key is an external/XML ID, set 'odoo_unique_key_is_xmlid' to True
+    # if this key is an external/XML ID, prefix the name with `xid::` (eg: xid::id)
     odoo_unique_key = ""
-    odoo_unique_key_is_xmlid = False
```
Comment (sebalix) [inline connector_importer/components/importer.py:69]: This is a breaking change for existing implementation. Should we add a warning somewhere?

Comment (simahawk) [reply connector_importer/components/importer.py:69]: all odoo_unique_key attributes are set to id which is already taken as an XMLID so all good

**Pull request merged**